qark/scanner/scanner.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from os import (
+    walk,
+    path
+)
+
+from qark.plugins.helpers import get_min_sdk, get_target_sdk
+from qark.scanner.plugin import get_plugin_source, get_plugins
+
+log = logging.getLogger(__name__)
+
+
+class Scanner(object):
+    __instance = None
+
+    def __new__(cls, decompiler):
+        if Scanner.__instance is None:
+            Scanner.__instance = object.__new__(cls)
+            Scanner.__instance.files = set()
+            Scanner.__instance.issues = set()
+
+        Scanner.__instance.decompiler = decompiler
+        return Scanner.__instance
+
+    def __init__(self, decompiler):
+        """
+        Creates the scanner.
+        :param Decompiler decompiler: the decompiler class that contains decompiled path information
+        """
+        self.decompiler = decompiler
+        if self.decompiler.manifest_path is None:
+            self.decompiler.manifest_path = self.decompiler.run_apktool()
+
+    def run(self):
+        """
+        Runs all the plugin checks by category.
+        """
+        self._gather_files()
+        self._run_manifest_checks()
+
+    def _run_manifest_checks(self):
+        """
+        Runs all plugins under `qark.plugins.manifest` and updates `self.issues` with their findings.
+        """
+        plugin_source = get_plugin_source(category="manifest")
+        for plugin_name in get_plugins(category="manifest"):
+            try:
+                plugin = plugin_source.load_plugin(plugin_name).plugin
+            except Exception:
+                log.exception("Error loading plugin %s... continuing with next plugin", plugin_name)
+                continue
+
+            plugin.run(files=self.files, extras={"minimum_sdk": get_min_sdk(self.decompiler.manifest_path),
```

Comment:
this should be surrounded in a try except

Reply:
Fixed.